pkg/kv/kvserver/replica_command.go

```diff
@@ -3157,7 +3163,30 @@ func (r *Replica) followerSendSnapshot(
 	// a snapshot for a non-system range. This allows us to send metadata of
 	// sstables in shared storage as opposed to streaming their contents. Keys
 	// in higher levels of the LSM are still streamed in the snapshot.
-	sharedReplicate := r.store.cfg.SharedStorageEnabled && snap.State.Desc.StartKey.AsRawKey().Compare(keys.TableDataMin) >= 0
+	nonSystemRange := snap.State.Desc.StartKey.AsRawKey().Compare(keys.TableDataMin) >= 0
+	sharedReplicate := r.store.cfg.SharedStorageEnabled && nonSystemRange
+
+	// Use external replication if we aren't using shared
+	// replication, are dealing with a non-system range, are on at
+	// least 24.1, and our store has external files.
+	externalReplicate := !sharedReplicate && nonSystemRange &&
+		r.store.ClusterSettings().Version.IsActive(ctx, clusterversion.V24_1) &&
+		externalFileSnapshotting.Get(&r.store.ClusterSettings().SV)
+	if externalReplicate {
+		start := snap.State.Desc.StartKey.AsRawKey()
+		end := snap.State.Desc.EndKey.AsRawKey()
+		_, _, external, err := r.store.StateEngine().ApproximateDiskBytes(start, end)
+		if err != nil {
+			log.Warningf(ctx, "could not determine if store has external bytes: %v", err)
+			externalReplicate = false
+		}
+		// TODO(ssd): Perhaps we should have a threshold
+		// here. Since we have to download files on all
+		// replicas eventually anyway, if the external bytes
+		// are small, it could be worth it to just send them
+		// now.
+		externalReplicate = external > 0
```

Comment:
I'd put this as external bytes > local bytes instead of any nonzero external bytes. Seems like a better heuristic that we likely won't even need to revisit.

Reply:
Nice idea, I agree that is a useful heurstic.